python/feathub/processors/flink/flink_types_utils.py

```diff
@@ -92,6 +98,75 @@ def to_flink_type(feathub_type: DType) -> DataType:
     )
 
 
+def cast_field_type(
+    table: NativeFlinkTable, field_name_and_types: Dict[str, DataType]
+) -> NativeFlinkTable:
+    """
+    Converts given fields in the flink table to the dedicated data types.
+
+    Note that this method would not change the nullable information of the fields.
+
+    :param table: the Flink table containing the fields to be casted
+    :param field_name_and_types: a dict containing the names of the fields and the
+                                 target data types to cast.
+    """
+    expressions = []
+    for field_name, field_type in field_name_and_types.items():
+        current_type = table.get_schema().get_field_data_type(field_name)
+        result_type = _adjust_flink_type_nullable_information(field_type, current_type)
+        if result_type == current_type:
+            continue
+        expressions.append(
+            native_flink_expr.col(field_name).cast(result_type).alias(field_name)
+        )
+
+    if not expressions:
+        return table
+    return table.add_or_replace_columns(*expressions)
+
+
+def _adjust_flink_type_nullable_information(
+    original_type: DataType, reference_type: DataType
+) -> DataType:
+    """
+    :param original_type: DataType containing correct information other than whether
+                          the data is nullable.
+    :param reference_type: DataType containing the correct nullable information.
+    :return: a DataType with the correct type information, including nullable.
```

Comment:
Can you clarify what it means to have `correct type information`? It is not clear what this method actually does by reading this doc.